src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/ScikitLearnVariantTrain.java

```diff
@@ -0,0 +1,456 @@
+            optional=true)
+    private List<String> IGNORE_INPUT_FILTERS = new ArrayList<>();
+
+    @Argument(fullName="ignore-all-filters",
+            doc="If specified, the variant recalibrator will ignore all input filters. Useful to rerun the VQSR from a filtered output file.",
+            optional=true)
+    private boolean IGNORE_ALL_FILTERS = false;
+
+    @Argument(fullName="python-script")
+    private File pythonScriptFile = null;
+
+    /**
+     * This argument is intended to be used in a more complicated VQSR scheme meant for very large WGS callsets that
+     * require a prohibitive amount of memory for classic VQSR. Given that training data is downsampled once it exceeds
+     * --max-num-training-data, reading in additional data to build the model only serves to consume resources. However,
+     * with this argument the output recal file will also be downsampled. The recommended VQSR procedure when using this
+     * argument is to run VariantRecalibrator once with sampling and designate an --output-model file. Then
+     * VariantRecalibrator can be run a second time scattered using the -scatterTranches argument and that file as an
+     * --input-model.  The scattered recal files can be gathered with the GatherVcfs tool and the scattered tranches can
+     * be gathered with the GatherTranches tool.
+     *
+     */
+    @Argument(fullName="sample-every-Nth-variant",
+            shortName = "sample-every",
+            doc="If specified, the variant recalibrator will use (and output) only a subset of variants consisting of every Nth variant where N is specified by this argument; for use with --output-model -- see argument details",
+            optional=true)
+    @Hidden
+    private int sampleMod = 1;
+
+    /**
+     * Add VQSLOD slices through the call set at the given values, to be used with the -scatterTranches argument. The
+     * default values span from -10 to +10 at varying resolution. The resolution output here will affect the accuracy
+     * with which the gathered tranches are able to match the requested truth sensitivity levels.
+     *
+     * Alternative lists of tranche values should only be used for testing.
+     */
+    @Hidden
+    @Argument(fullName="vqslod-tranche",
+            doc="The levels of VQSLOD at which to slice the data.",
+            optional=true)
+    private List<Double> VQSLOD_TRANCHES = new ArrayList<>(1000);
+    {
+        for (double i=10.0; i>5; i-=0.1) {
+            VQSLOD_TRANCHES.add(i);
+        }
+        for (double i=5.0; i>-5; i-=0.01) {
+            VQSLOD_TRANCHES.add(i);
+        }
+        for (double i=-5.0; i>-10; i-=0.1) {
+            VQSLOD_TRANCHES.add(i);
+        }
+    };
+
+    /**
+     * The statistical model being built by this tool may fail due to simple statistical sampling
+     * issues. Rather than dying immediately when the initial model fails, this argument allows the
```

Comment:
What exactly is the failure?

Reply:
The VQSR GMM implementation is extremely prone to numerical issues involving the covariance matrices (positive semidefiniteness, etc.). There were some attempts to add regularization bandaids over the years, but these did not sufficiently address the issues.